backend/src/handlers/middleware.go

```diff
@@ -62,19 +70,54 @@ func (srv *Server) libraryProxyMiddleware(next http.Handler) http.Handler {
 			http.Redirect(w, r, AuthCallbackRoute, http.StatusSeeOther)
 			return
 		}
+		libraryBucket := srv.buckets[LibraryPaths]
 		resourceID := r.PathValue("id")
+		entry, err := libraryBucket.Get(resourceID)
+		var proxyParams models.LibraryProxyPO
+		if err == nil { //found in bucket going to use cached slice of bytes
+			err = json.Unmarshal(entry.Value(), &proxyParams)
+			if err != nil {
+				log.Warnf("issue unmarshaling LibraryProxyPO from nats bucket, error is: %v", err)
+			}
+		}
 		var library models.Library
-		tx := srv.Db.Model(&models.Library{}).Preload("OpenContentProvider").Where("id = ?", resourceID)
-		if user.isAdmin() {
-			tx = tx.First(&library)
-		} else {
-			tx = tx.First(&library, "visibility_status = true")
+		if err != nil { //check if there was an error
+			if srv.Db.Debug().Model(&models.Library{}).Preload("OpenContentProvider").Where("id = ?", resourceID).First(&library).RowsAffected == 0 {
+				srv.errorResponse(w, http.StatusNotFound, "Library not found.")
+				return
+			}
+			proxyParams = models.LibraryProxyPO{
```

Comment:
hey can we make a method on `models.Library` called `IntoProxyP0`, just so we can bring this down to 1 line?

Reply:
Good call, made the change, added IntoProxyPO method to Library and made the necessary changes.